### PR TITLE
Share timer implementation

### PR DESCRIPTION
**Pull request by acoates-ms**
We have two implementations of the TimingModule in the repo.  One that we use in Office and one used in XAML apps.  With this change we will move to the implementation that does not depend on XAML functions, which also makes it easier for people hooking RNW to run against a different UI platforms, to no longer need to provide a custom Timer module.

Since its likely that replacing the TimingModule could have weird affects on apps, I've provided a quirk setting to revert back to the old implementation until we get confirmation that the new one doesn't cause undue regressions.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/10338)

### CODE REVIEW

**Review by vmoroz — APPROVED** [review]
:shipit:

File: vnext/Microsoft.ReactNative/Modules/TimingModule.h [hunk 1]
```diff
@@ -3,96 +3,116 @@
...
 #include <vector>
 
-#include <winrt/Windows.Foundation.h>
-namespace Microsoft::ReactNative {
+#include <windows.h>
 
-typedef winrt::Windows::Foundation::DateTime TDateTime;
-typedef winrt::Windows::Foundation::TimeSpan TTimeSpan;
+namespace facebook {
+namespace react {
```
Comment (asklar) [inline vnext/Microsoft.ReactNative/Modules/TimingModule.h:17]: why is this in the facebook::react namespace?

**Review by asklar — CHANGES_REQUESTED** [review]
> Since its likely that replacing the TimingModule could have weird affects on apps, I've provided a quirk setting to revert back to the old implementation until we get confirmation that the new one doesn't cause undue regressions.

I think we would want the quirk to work the other way by default, if it's not set, we should default to what already works for UWP apps, until we have confirmation it doesn't break anything

File: vnext/Microsoft.ReactNative/Modules/TimingModule.h (hunk 1 above)
Comment (acoates-ms) [reply vnext/Microsoft.ReactNative/Modules/TimingModule.h:17]: Its currently being exported from the desktop dll, so I didn't want to change the name.  But it doesn't look like we are using it anymore, so I'll put it in the MS.RN namespace, and remove it from the def files.

**Comment by acoates-ms** [comment]
> > Since its likely that replacing the TimingModule could have weird affects on apps, I've provided a quirk setting to revert back to the old implementation until we get confirmation that the new one doesn't cause undue regressions.
> 
> I think we would want the quirk to work the other way by default, if it's not set, we should default to what already works for UWP apps, until we have confirmation it doesn't break anything

I dont think we'll get much in app test coverage with it off by default.  Most apps are not going to go through and do the extra work to enable it.

**Comment by NickGerleman** [comment]
IIRC the Office timer doesn't have correct semantics to support `requestAnimationFrame()` as polyfilled by react-native. https://github.com/facebook/react-native/blob/1af2beaaa34ac01f04b14035acdfe0a5ecb329d8/Libraries/Core/Timers/JSTimers.js#L259

See https://github.com/microsoft/react-native-windows/issues/6032 and Eric's fix for some more context.

**Comment by rozele** [comment]
+1 @NickGerleman. I'm all for ridding the implementation of DispatcherTimer, but all other platforms fire RAF callbacks on frame boundaries, so I'd be concerned that this would break JS -driven Animated xplat consistency. Given the bugs in NativeAnimated, we should ensure those work properly.

**Comment by chiaramooney** [comment]
ping @vmoroz

**Comment by jonthysell** [comment]
@acoates-ms Can you put this as a draft?

**Comment by acoates-ms** [comment]
Timers have been refactored since bridgeless.  This change is no longer needed

**Pull request closed without merging**